test/e2e/util_test.go

```diff
@@ -688,3 +689,89 @@ func assertDeletedWaitForCleanup(t *testing.T, cl client.Client, thing client.Ob
 		t.Fatalf("Timed out waiting for %s to be cleaned up: %v", thing.GetName(), err)
 	}
 }
+
+// dumpEventsInNamespace gets the events in the specified namespace and logs
+// them.
+func dumpEventsInNamespace(t *testing.T, ns string) {
+	t.Helper()
+
+	eventList := &corev1.EventList{}
+	if err := kclient.List(context.TODO(), eventList, client.InNamespace(ns)); err != nil {
+		t.Errorf("failed to list events for namespace %s: %v", ns, err)
+		return
+	}
+
+	for _, e := range eventList.Items {
+		t.Log(e.FirstTimestamp, e.Source, e.InvolvedObject.Kind, e.InvolvedObject.Name, e.Reason, e.Message)
+	}
+}
+
+// createNamespace creates a namespace with the specified name and registers a
+// cleanup handler to delete the namespace when the test finishes.
+//
+// After creating the namespace, this function waits for the "default"
+// ServiceAccount and "system:image-pullers" RoleBinding to be created as well,
+// which is necessary in order for pods in the new namespace to be able to pull
+// images.
+func createNamespace(t *testing.T, name string) *corev1.Namespace {
+	t.Helper()
+
+	t.Logf("Creating namespace %q...", name)
+	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: name}}
+	if err := kclient.Create(context.TODO(), ns); err != nil {
+		t.Fatalf("failed to create namespace: %v", err)
+	}
+	t.Cleanup(func() {
+		t.Logf("Dumping events in namespace %q...", name)
```

Comment:
I hadn't noticed this earlier, but this log message belongs inside the `if t.Failed()` condition brackets.  I don't think it needs to fixed in the other PRs, but do you want to fix it here and in the 4.11 backports while they are open?

Reply:
> I hadn't noticed this earlier, but this log message belongs inside the `if t.Failed()` condition brackets.

According to https://pkg.go.dev/testing#T.Log:

> For tests, the text will be printed only if the test fails or the -test.v flag is set.

So I don't believe there's a need to add `if t.Failed()`.

> I don't think it needs to fixed in the other PRs, but do you want to fix it here and in the 4.11 backports while they are open?

Backports should be as close as possible to the original changes (though resolving conflicts sometimes necessitates some deviation).  Otherwise it gets more difficult to keep track of what actually got backported to which version, and subsequent backports are more likely to encounter conflicts.  If we need additional changes to this code, I'd rather open a new Jira issue, make the additional change in the main branch, and do another series of backports if necessary.

Reply:
I'll leave that up to you, it's not a big deal.